### PR TITLE
Do not trigger error when deleting default account

### LINKED ISSUE #302: Deleting the default account will trigger an error
STR:
- set an account as a default account
- delete it from the preferences
- reload page

So this has to be done on the server: when deleting an account, check if it is the default account and remove the default account value in that case.

@nicofrand, willing to fix this?

### PR DESCRIPTION
**Pull request by nicofrand**
Do not trigger an error when the account deleted is the default account
Fix #302 

[Review on Reviewable](https://reviewable.io/reviews/bnjbvr/kresus/304)

### CODE REVIEW

File: client/components/OperationList.js [hunk 1]
```diff
@@ -530,4 +530,3 @@ export default class OperationsComponent extends React.Component {
         );
     }
 }
-
```
Comment (nicofrand) [inline client/components/OperationList.js]: I do not understand why the empty last line is removed whereas I still have it on my local copy…

File: server/controllers/accounts.js [hunk 2]
```diff
@@ -33,6 +34,14 @@ export async function destroyWithOperations(account) {
     log.info(`\t-> Destroy alerts for account ${account.title}`);
     await Alert.destroyByAccount(account.accountNumber);
 
+    log.info(`\t-> Checking if ${account.title} is the default account`);
+    let found = await Config.findOrCreateByName('defaultAccountId', '');
+    if (found && (found.value === account.id)) {
```
Comment (bnjbvr) [inline server/controllers/accounts.js]: nit: needless parenthesis here around the `===` operator

**Comment by bnjbvr** [comment]
Looks good, thanks!

**Comment by bnjbvr** [comment]
Nice, cheers!

---

Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/304#-:-K6eheX2B3ihn5W13llg:138088549)*

**Pull request merged**